src/components/ShareResultsModal/ShareResultsModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import Modal from 'react-responsive-modal';
+import 'react-responsive-modal/styles.css';
+
+import CloseSVG from './CloseSVG';
+import LinkField from './LinkField';
+import EmailField from './EmailField';
+
+import { QuizReturnState } from '../../types';
+
+export default function ShareResultsModal({
+  showShareModal,
+  onClose,
+  quizState,
+  onEmailResults,
+  containerRef,
+}: Props) {
+  const urlObj = new URL(window.location.href);
+  const existingParams = urlObj.searchParams;
+  existingParams.set(
+    'items',
+    quizState.results?.response?.results
+      .filter((item) => item.data?.id)
+      .map((item) => item.data!.id)
+      .join(',') || ''
+  );
+  existingParams.set(
+    'attributes',
+    quizState.selectedOptionsWithAttributes?.map((option) => option).join(',') || ''
+  );
+  const value = urlObj.toString();
```

Comment:
Can we make this functionality available for the hooks interface as well? Like if people are using the hooks interface they won't have access to this feature. Depending on how quick we want this out I will leave it to you if you agree with me to decide whether to refactor this in this PR to be into a hook (something like useShareResultsLink() or do it in another PR.

Reply:
Oh good call! That won't be hard to move 👍 d280bd3f39a75f56bc6e87fe8e9363857411cf1e let me know if this is what you intended though. Also, do I have to update documentation or ts docs?

Reply:
Oh good call! That won't be hard to move 👍 d280bd3f39a75f56bc6e87fe8e9363857411cf1e let me know if this is what you intended though. Also, do I have to update documentation or ts docs?

Reply:
Thanks for adding this. We will follow up with a usage example after the new docs PR gets merged

Reply:
We will follow up with a usage example after the new docs PR gets merged

Reply:
Thanks for refactoring